src/middleware/auth.js

```
@@ -0,0 +1,43 @@
+// src/middleware/auth.js
+const { verifyAccessToken, rotateRefreshToken, issueAccessToken } = require('../services/tokenService');
+
+/**
+ * Auth middleware:
+ * - Tenta validar o access token (header Bearer ou cookie)
+ * - Se inválido/ausente, tenta refresh cookie no DB:
+ *    - Se ok, rotaciona refresh, emite novo access, setta cookies e continua
+ *    - Se falhar, 401
+ */
+module.exports = async function auth(req, res, next) {
+  try {
+    const bearer = req.header('Authorization')?.replace('Bearer ', '');
+    const cookieAT = req.cookies?.accessToken;
+    const token = bearer || cookieAT;
+
+    if (token) {
+      const payload = verifyAccessToken(token);
```

If an access token exists but is invalid/expired, `verifyAccessToken(token)` will throw and the outer try/catch will return 401 immediately, skipping the refresh fallback. The middleware comment says it should try refresh on invalid/absent access tokens. Wrap the verification in its own try/catch and on verification failure continue to the refresh logic instead of returning 401 here.